File: src/p_3_creational_patterns/p_8_factoryabstract/IAbstractFactory.java
package p_3_creational_patterns.p_8_factoryabstract;

import p_3_creational_patterns.p_8_factoryabstract.bank.BankTypeEnum;
import p_3_creational_patterns.p_8_factoryabstract.bank.IBaseBank;
import p_3_creational_patterns.p_8_factoryabstract.loan.IBaseLoan;
import p_3_creational_patterns.p_8_factoryabstract.loan.LoanTypeEnum;

public interface IAbstractFactory {
    IBaseLoan getLoan(LoanTypeEnum type);
    IBaseBank getBank(BankTypeEnum type);
}

File: Net/03_creational_patterns/04_factoryabstract_pattern/IAbstractFactory.cs
using _04_factoryabstract_pattern.Bank;
using _04_factoryabstract_pattern.Loan;

namespace _04_factoryabstract_pattern
{
    public interface IAbstractFactory
    {
        ILoan getLoan(LoanTypes type);
        IBank getBank(BankTypes type);
    }
}

